runtime/command/run/run.go

```diff
@@ -2,75 +2,132 @@ package run
 
 import (
 	"context"
-	"net"
+	"errors"
+	"io/fs"
+	"os"
 	"path/filepath"
 
+	"github.com/livebud/bud/internal/buildcache"
+	"github.com/livebud/bud/internal/envs"
+	"github.com/livebud/bud/internal/extrafile"
 	"github.com/livebud/bud/package/exe"
+	"github.com/livebud/bud/package/gomod"
 	"github.com/livebud/bud/package/log/console"
+	"github.com/livebud/bud/package/overlay"
+	"github.com/livebud/bud/package/socket"
 	"github.com/livebud/bud/package/watcher"
 
 	"github.com/livebud/bud/package/hot"
 
 	"golang.org/x/sync/errgroup"
 
-	"github.com/livebud/bud/package/socket"
-	"github.com/livebud/bud/runtime/bud"
+	"github.com/livebud/bud/runtime/command"
+	"github.com/livebud/bud/runtime/web"
 )
 
+func New(module *gomod.Module) *Command {
+	return &Command{
+		module: module,
+		// Default flags
+		Flag: &command.Flag{
+			Embed:  false,
+			Hot:    ":35729",
+			Minify: false,
+		},
+	}
+}
+
+// Command to run the project at runtime
 type Command struct {
-	Flag    *bud.Flag
-	Project *bud.Project
-	Port    string
+	module *gomod.Module
+	// Below are filled in by the CLI
+	FS     *overlay.FileSystem
+	Flag   *command.Flag
+	Listen string
 }
 
 func (c *Command) Run(ctx context.Context) error {
 	eg, ctx := errgroup.WithContext(ctx)
-	// Initialize the hot server
-	var hotServer *hot.Server
-	if c.Flag.Hot {
-		hotServer = hot.New()
-		// Start the hot reload server
-		eg.Go(func() error { return c.startHot(ctx, hotServer) })
-	}
-	// Start the web server
+	// Start the live reload server
+	hotServer := hot.New()
+	eg.Go(func() error { return c.startHot(ctx, hotServer) })
+	// Start the app server
 	eg.Go(func() error { return c.startApp(ctx, hotServer) })
-	return eg.Wait()
+	// Wait until either the hot or web server exists
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+	return nil
 }
 
-func (c *Command) compileAndStart(ctx context.Context, ln net.Listener) (*exe.Cmd, error) {
-	app, err := c.Project.Compile(ctx, c.Flag)
-	if err != nil {
+func (c *Command) compileAndStart(ctx context.Context, listener socket.Listener) (*exe.Cmd, error) {
```

Comment:
Moved most of the `bud {run,build,...}` logic into the runtime to prep for wrapping built-in commands.